src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -113,125 +137,269 @@ internal abstract class AbstractAddParameterCodeFixProvider<
+            // TODO: Insert hint in the fix with a warning if anySymbolsReferenceNotInSource is true
+            var anySymbolReferencesNotInSource = referencedSymbols.Any(symbol => !symbol.IsFromSource());
+            var locationsInSource = referencedSymbols.Where(symbol => symbol.IsFromSource());
+            // Indexing Locations[0] is valid because IMethodSymbols have one location at most and IsFromSource() tests if there is at least one location.
+            var locationsByDocument = locationsInSource.ToLookup(declarationLocation
+                => solution.GetDocument(declarationLocation.Locations[0].SourceTree));
+            foreach (var documentLookup in locationsByDocument)
+            {
+                var document = documentLookup.Key;
+                var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+                var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var editor = new SyntaxEditor(syntaxRoot, solution.Workspace);
+                var generator = editor.Generator;
+                foreach (var methodDeclaration in documentLookup)
+                {
+                    var methodNode = syntaxRoot.FindNode(methodDeclaration.Locations[0].SourceSpan);
+                    var parameterSymbol = CreateParameterSymbol(
+                        methodDeclaration, argumentType, argumentNameSuggestion);
 
-            var (parameterSymbol, isNamedArgument) = await CreateParameterSymbolAsync(
-                invocationDocument, method, argument, cancellationToken).ConfigureAwait(false);
+                    var parameterDeclaration = generator.ParameterDeclaration(parameterSymbol)
+                                                        .WithAdditionalAnnotations(Formatter.Annotation);
+                    if (anySymbolReferencesNotInSource && methodDeclaration == method)
+                    {
+                        parameterDeclaration = parameterDeclaration.WithAdditionalAnnotations(
+                            ConflictAnnotation.Create(FeaturesResources.Related_method_signatures_found_in_metadata_will_not_be_updated));
+                    }
+                    var existingParameters = generator.GetParameters(methodNode);
+                    var insertionIndex = isNamedArgument
+                        ? existingParameters.Count
+                        : argumentList.IndexOf(argument);
 
-            var methodDocument = invocationDocument.Project.Solution.GetDocument(methodDeclaration.SyntaxTree);
-            var syntaxFacts = methodDocument.GetLanguageService<ISyntaxFactsService>();
-            var methodDeclarationRoot = methodDeclaration.SyntaxTree.GetRoot(cancellationToken);
-            var editor = new SyntaxEditor(methodDeclarationRoot, methodDocument.Project.Solution.Workspace);
+                    if (method.IsExtensionMethod)
+                    {
+                        insertionIndex++;
```

Comment:
good catch.

Reply:
Note: do you have tests where people call the method both in 'instance' form and 'static' form?  this may break if they originallly called it in static form.